modules/core/keeper/msg_server.go

```diff
@@ -737,6 +737,10 @@ func (k Keeper) Acknowledgement(goCtx context.Context, msg *channeltypes.MsgAckn
 func (k Keeper) ChannelUpgradeInit(goCtx context.Context, msg *channeltypes.MsgChannelUpgradeInit) (*channeltypes.MsgChannelUpgradeInitResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
+	if k.GetAuthority() != msg.Signer {
+		return nil, errorsmod.Wrapf(ibcerrors.ErrUnauthorized, "expected signer address to be the same as the authority address %s, got %s", k.GetAuthority(), msg.Signer)
```

Comment:
Maybe we should have a consistent error message for this check? https://github.com/cosmos/ibc-go/blob/c694852e20e8191fd49ad3cfd4b25208b78e5b1c/modules/core/keeper/msg_server.go#L106

Reply:
actually, it looks like theres an invalid signer error in `govtypes`. maybe we can update all of these to use that error return, looks like that's what the 08-wasm client returns in `StoreCode`

Reply:
sounds good, maybe we open an issue and address once all the feature branches are merged to main?